### DISCUSSION TITLE
how to change the position of expandIcon in Menu?

### DISCUSSION THREAD

**Discussion by salekin02** [ant-design/ant-design#52045] [Q&A]
I want to show the expandIcon at left side not at end of the title. Any way?

https://ant.design/components/menu#menu-demo-switch-mode

**Comment by petercat-assistant** [comment]
To change the position of the `expandIcon` in the Ant Design `Menu` component, you can customize the rendering of the `expandIcon` by using the `expandIcon` prop. This prop allows you to define a custom icon component, and you can position it as needed.

Here's a basic example of how you can achieve this:

```jsx
import React from 'react';
import { Menu } from 'antd';
import { DownOutlined } from '@ant-design/icons';

const { SubMenu } = Menu;

const CustomMenu = () => {
  return (
    <Menu mode="inline">
      <SubMenu
        key="sub1"
        title={
          <span>
            <DownOutlined style={{ marginRight: 8 }} />
            Navigation One
          </span>
        }
      >
        <Menu.Item key="1">Option 1</Menu.Item>
        <Menu.Item key="2">Option 2</Menu.Item>
      </SubMenu>
    </Menu>
  );
};

export default CustomMenu;
```

In this example, the `DownOutlined` icon is used as the `expandIcon`, and it's placed at the beginning of the `title` span. You can replace `DownOutlined` with any other icon or custom component you prefer.

If you need further customization or run into any issues, feel free to ask!

<details>
<summary>🪧 Tips</summary>
For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.
</details>

**Reply by salekin02** [reply]
in that case how can I change the icon based on isOpen of the menu item dynamically?

**Comment by salekin02** [comment]
@petercat-assistant in that case how can I change the icon based on isOpen of the menu item dynamically?

**Comment by petercat-assistant** [comment]
To change the icon dynamically based on whether the menu item is open or closed, you can use the `openKeys` state of the `Menu` component to determine the current open state of the submenu. You can then conditionally render the icon based on this state.

Here's an example of how you can achieve this:

```jsx
import React, { useState } from 'react';
import { Menu } from 'antd';
import { DownOutlined, RightOutlined } from '@ant-design/icons';

const { SubMenu } = Menu;

const DynamicIconMenu = () => {
  const [openKeys, setOpenKeys] = useState([]);

  const onOpenChange = (keys) => {
    setOpenKeys(keys);
  };

  return (
    <Menu
      mode="inline"
      openKeys={openKeys}
      onOpenChange={onOpenChange}
    >
      <SubMenu
        key="sub1"
        title={
          <span>
            {openKeys.includes('sub1') ? <DownOutlined style={{ marginRight: 8 }} /> : <RightOutlined style={{ marginRight: 8 }} />}
            Navigation One
          </span>
        }
      >
        <Menu.Item key="1">Option 1</Menu.Item>
        <Menu.Item key="2">Option 2</Menu.Item>
      </SubMenu>
    </Menu>
  );
};

export default DynamicIconMenu;
```

In this example, the `openKeys` state is used to track which submenu is open. The `onOpenChange` function updates this state whenever a submenu is opened or closed. The icon is conditionally rendered based on whether the submenu key is included in the `openKeys` array.

- `DownOutlined` is shown when the submenu is open.
- `RightOutlined` is shown when the submenu is closed.

You can replace these icons with any other icons or custom components as needed.

<details>
<summary>🪧 Tips</summary>
For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.
</details>
